src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1289,4 +1290,1373 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+    } else if (!this->sourceSideDesignVolFlowRateWasAutoSized && this->sourceSideDesignVolFlowRate == 0.0) { // LCOV_EXCL_LINE
+        // user gave a flow rate of 0
+        // protected by the input processor to be >0.0
+        // fatal out just in case
+        errorsFound = true; // LCOV_EXCL_LINE
+        ShowSevereError(state,
+                        format("Invalid condenser flow rate for EIR PLHP (name={}; entered value: {}",
+                               this->name,
+                               this->sourceSideDesignVolFlowRate)); // LCOV_EXCL_LINE
+    } else {
+        // can't imagine how it would ever get to this point
+        // just assume it's the same as the load side if we don't have any sizing information
+        tmpSourceVolFlow = tmpLoadVolFlow; // LCOV_EXCL_LINE
+    }
+
+    this->sourceSideDesignVolFlowRate = tmpSourceVolFlow;
+
+    if (errorsFound) {
+        ShowFatalError(state, "Preceding sizing errors cause program termination"); // LCOV_EXCL_LINE
+    }
+}
+
+void EIRFuelFiredHeatPump::resetReportingVariables()
+{
+    this->loadSideHeatTransfer = 0.0;
+    this->loadSideEnergy = 0.0;
+    this->loadSideOutletTemp = this->loadSideInletTemp;
+    this->powerUsage = 0.0;
+    this->powerEnergy = 0.0;
+    this->fuelUsage = 0.0;
+    this->fuelEnergy = 0.0;
+    this->sourceSideHeatTransfer = 0.0;
+    this->sourceSideOutletTemp = this->sourceSideInletTemp;
+    this->sourceSideEnergy = 0.0;
+}
+
+PlantComponent *EIRFuelFiredHeatPump::factory(EnergyPlusData &state, DataPlant::PlantEquipmentType hp_type_of_num, const std::string &hp_name)
+{
+    if (state.dataEIRFuelFiredHeatPump->getInputsFFHP) {
+        EIRFuelFiredHeatPump::processInputForEIRPLHP(state);
+        EIRFuelFiredHeatPump::pairUpCompanionCoils(state);
+        state.dataEIRFuelFiredHeatPump->getInputsFFHP = false;
+    }
+
+    for (auto &plhp : state.dataEIRFuelFiredHeatPump->heatPumps) {
+        if (plhp.name == UtilityRoutines::MakeUPPERCase(hp_name) && plhp.EIRHPType == hp_type_of_num) {
+            return &plhp;
+        }
+    }
+
+    ShowFatalError(state, "EIR Fuel-Fired Heat Pump factory: Error getting inputs for PLFFHP named: " + hp_name);
+    return nullptr; // LCOV_EXCL_LINE
+}
+
+void EIRFuelFiredHeatPump::pairUpCompanionCoils(EnergyPlusData &state)
+{
+    for (auto &thisHP : state.dataEIRFuelFiredHeatPump->heatPumps) {
+        if (!thisHP.companionCoilName.empty()) {
+            auto thisCoilName = UtilityRoutines::MakeUPPERCase(thisHP.name);
+            auto &thisCoilType = thisHP.EIRHPType;
+            auto targetCompanionName = UtilityRoutines::MakeUPPERCase(thisHP.companionCoilName);
+            for (auto &potentialCompanionCoil : state.dataEIRFuelFiredHeatPump->heatPumps) {
+                auto &potentialCompanionType = potentialCompanionCoil.EIRHPType;
+                auto potentialCompanionName = UtilityRoutines::MakeUPPERCase(potentialCompanionCoil.name);
```

Comment:
Should probably just store both `name` and `nameUC` so that you are not uppercasing multiple times. This is true for all EnergyPlus objects BTW.

Reply:
> OK, I'm testing this locally and things seem good. However, it looks like there are several comments that were just never responded to. @jcyuan2020 I'd like to get this in next, and I'd feel better if those were at least collected into a follow-up issue, or addressed here before merge.

Just pushed some recent changes. I am sifting through the comments over again.

Reply:
> 

Will an additional "name" solve the problem? Actually there would be all kinds of strange combinations for each character to be capital or lower cases. So it looks for be ultimately certain, case conversions have to be made for all upper-case (or all-lower case which we are not widely using) each time.